pkg/proxy/conntrack/cleanup.go

```diff
@@ -169,31 +168,31 @@ var protocolMap = map[v1.Protocol]uint8{
 // filterForIPPortNAT returns *conntrackFilter to delete the conntrack entries for connections
 // specified by the destination IP (original direction) and destination port (original direction)
 // and source IP (reply direction).
-func filterForIPPortNAT(origin, dest string, dstPort uint16, protocol v1.Protocol) *conntrackFilter {
-	klog.V(6).InfoS("Adding conntrack filter for cleanup", "org-dst", origin, "reply-src", dest, "protocol", protocol)
+func filterForIPPortNAT(origDst net.IP, origPortDst uint16, replySrc net.IP, protocol v1.Protocol) *conntrackFilter {
+	klog.V(6).InfoS("Adding conntrack filter for cleanup", "orig-dst", origDst.String(), "orig-port-dst", origPortDst, "reply-src", replySrc.String(), "protocol", protocol)
 	return &conntrackFilter{
 		protocol: protocolMap[protocol],
 		original: &connectionTuple{
-			dstIP:   netutils.ParseIPSloppy(origin),
-			dstPort: dstPort,
+			dstIP:   origDst,
+			dstPort: origPortDst,
 		},
 		reply: &connectionTuple{
-			srcIP: netutils.ParseIPSloppy(dest),
+			srcIP: replySrc,
```

Comment:
Yay! Every time you call `ParseIPSloppy()` on an IP you just `.String()`ed, God kills a kitten.